nixos/modules/services/misc/tiddlywiki.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.tiddlywiki;
+  listenParams = concatStrings (mapAttrsToList (n: v: " '${n}=${toString v}' ") cfg.listenOptions);
+  package = pkgs.nodePackages.tiddlywiki;
+  exe = "${package}/lib/node_modules/.bin/tiddlywiki";
+
+in {
+
+  options.services.tiddlywiki = {
+
+    enable = mkEnableOption "TiddlyWiki nodejs server";
+
+    wikipath = mkOption {
+      type = types.path;
+      default = "/var/lib/tiddlywiki";
+      description = "Path to wiki folder.";
+    };
+
+    listenOptions = mkOption {
+      type = types.attrs;
+      default = {};
+      example = {
+        credentials = "../credentials.csv";
+        readers="(authenticated)";
+        port = 3456;
+      };
+      description = "Parameters passed to '--lisen' command.";
```

Comment:
Is there a link to documentation that can be included here? See https://nixos.org/nixos/options.html#services.davmail.config source for a nice example.

Reply:
added link to documentation